### PR TITLE
Ported TheXaman's latest changes to the Debug Menu

### LINKED ISSUE #2821: "Fill PC" option gives all mon Bulbasaur's moves
mGBA_lodG2GC3v6

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
### Features:
- [ ] ~Option to generate a test battle, reading from the boxes.~ Commented out for now, battles working, but not reading data from boxes.
- [x] Arrows and ellipsis mark submenus.
- [x] Added 9th scrolling option to occupy the whole screen height.
- [x] Combine Flags and Vars into one submenu.
  - [x] Moved Running Shoes flag to this menu.
  - [x] Added new window to flags/vars showing the current state and added submenu indicator.
  - [x] Colored toggle options for specific flags that change upon toggling.
  - [x] Added option to reset Pokédex flags to whatever is in the party and boxes.
- [x] New "Fill PC/Item Pocket" submenu for filling both PC and Bag Pockets.
  - [x] Option to generate Box Mon with their own personalities/IVs.
  - [x] Fix #2821.
- [x] Moved "CHEAT Start" option to the Utility Submenu"

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Finally ready for review.

File: src/debug.c [hunk 1]
```diff
@@ -432,62 +589,108 @@ static const s32 sPowersOfTen[] =
...
-    [DEBUG_FLAG_MENU_ITEM_FRONTIER_PASS]     = {sDebugText_Flags_ToggleFrontierPass, DEBUG_FLAG_MENU_ITEM_FRONTIER_PASS},
-    [DEBUG_FLAG_MENU_ITEM_COLISSION_ONOFF]   = {sDebugText_Flags_SwitchCollision,    DEBUG_FLAG_MENU_ITEM_COLISSION_ONOFF},
-    [DEBUG_FLAG_MENU_ITEM_ENCOUNTER_ONOFF]   = {sDebugText_Flags_SwitchEncounter,    DEBUG_FLAG_MENU_ITEM_ENCOUNTER_ONOFF},
-    [DEBUG_FLAG_MENU_ITEM_TRAINER_SEE_ONOFF] = {sDebugText_Flags_SwitchTrainerSee,   DEBUG_FLAG_MENU_ITEM_TRAINER_SEE_ONOFF},
-    [DEBUG_FLAG_MENU_ITEM_BAG_USE_ONOFF]     = {sDebugText_Flags_SwitchBagUse,       DEBUG_FLAG_MENU_ITEM_BAG_USE_ONOFF},
-    [DEBUG_FLAG_MENU_ITEM_CATCHING_ONOFF]    = {sDebugText_Flags_SwitchCatching,     DEBUG_FLAG_MENU_ITEM_CATCHING_ONOFF},
+static const struct ListMenuItem sDebugMenu_Items_Battle_0[] =
+{
+    [DEBUG_BATTLE_0_MENU_ITEM_WILD]        = {sDebugText_Battle_0_Wild,       DEBUG_BATTLE_0_MENU_ITEM_WILD},
+    #ifdef BATTLE_ENGINE
```
Comment (DizzyEggg) [inline src/debug.c:652]: what does BATTLE_ENGINE mean here? Is it a leftover from 3 branches?

File: src/wild_encounter.c [hunk 2]
```diff
@@ -1113,3 +1113,12 @@ bool8 TryDoDoubleWildBattle(void)
...
     return FALSE;
 }
+
+bool8 StandardWildEncounter_Debug(void)
+{
+    u16 headerId = GetCurrentMapWildMonHeaderId();
+    if (TryGenerateWildMon(gWildMonHeaders[headerId].landMonsInfo, WILD_AREA_LAND, 0) != TRUE)
+        return FALSE;
+
+    DoStandardWildBattle_Debug();
```
Comment (DizzyEggg) [inline src/wild_encounter.c:1123]: It's missing return TRUE at the end of the function.

File: src/debug.c (hunk 1 above)
Comment (AsparagusEduardo) [reply src/debug.c:652]: yeah, will remove.

**Comment by AsparagusEduardo** [comment]
Review changes applied.

**Comment by DizzyEggg** [comment]
Any reason Fill PC Boxes Fast sets all mons' nicknames to Bulbasaur?

**Comment by AsparagusEduardo** [comment]
> Any reason Fill PC Boxes Fast sets all mons' nicknames to Bulbasaur?

The current "Fill PC" option generates data for Bulbasaur and then copies it across boxes while changing the species.

"Fill PC Boxes Fast" is basically the current "Fill PC" option, but with the movesets reassigned.
All this to say that I forgor 💀 , will fix ^^;

**Comment by DizzyEggg** [comment]
By the way, do you want this merged now or after the last task is finished?
> Option to generate a test battle, reading from the boxes. Commented out for now, battles working, but not reading data from boxes.

**Comment by AsparagusEduardo** [comment]
> By the way, do you want this merged now or after the last task is finished?

I think it's fine to do it now, I commented it because I haven't touched it for a while ^^;

**Review by DizzyEggg — CHANGES_REQUESTED** [review]
GetSpeciesName function was changed in upcoming. Please merge with upcoming and correct the calls to that function.

**Comment by AsparagusEduardo** [comment]
Done

**Review by DizzyEggg — APPROVED** [review]

**Pull request merged**